File: fizzbuzz.rb
# Calculates and returns an array of "FizzBuzz" numbers up to an entered value
# Divisible by 3 = "Fizz", 5 = "Buzz", 3 and 5 = "FizzBuzz"


module FizzBuzz

  extend self

  def solve value
    value = value.to_i
    fizz_nums = []
    1.upto value do |count|
      if count % 15 == 0
        fizz_nums << "FizzBuzz"
      elsif count % 5 == 0
        fizz_nums << "Buzz"
      elsif count % 3 == 0
        fizz_nums << "Fizz"
      else
        fizz_nums << count
      end
    end
    fizz_nums
  end

end

if $0 === __FILE__
  value = ARGV[0].to_i
  puts FizzBuzz.solve value
end

File: fizzbuzz.js
// Calculates and returns an array of "FizzBuzz" numbers up to an entered value
// Any values that are divisible by 3 = "Fizz", 5 = "Buzz", 3 and 5 = "FizzBuzz"

var FizzBuzz = {
  solve: function(value) {
    var result = []
    for(var count = 1; count <= value; count++) {
      if(count % 15 === 0) {
        result.push('FizzBuzz')
      }
      else if(count % 3 === 0) {
        result.push('Fizz')
      }
      else if(count % 5 === 0) {
        result.push('Buzz')
      }
      else {
        result.push(count.toString())
      }
    }
    return result
  }
}

FizzBuzz.solve(31)
